### PR TITLE
Fixed letters size problem and no common letters in text and frequency bug.

### PR DESCRIPTION
**Pull request by gdziadkiewicz**

### CODE REVIEW

File: LanguageRecognition/Engine.fs [hunk 1]
```diff
@@ -127,20 +127,19 @@ let getFrequencies (countedChars: seq<char*int>) =
...
 let transformFst transformation (x, y) = (transformation x, y)
 
-let checkCompatibility (similarity:Similarity) (languageFrequency: seq<char*float>) (textFrequency: seq<char*float>) = 
-    if Seq.isEmpty languageFrequency || Seq.isEmpty textFrequency then 0.
-    else
-        let f seq1 (key, _)  = seq1 |> Seq.map fst |> Seq.contains key
-        let textFrequencyVector = textFrequency |> Seq.where (f languageFrequency)|> Seq.sortBy fst |> Seq.map snd |> Vector.Build.SparseOfEnumerable
-        let languageFrequencyVector = languageFrequency |> Seq.map (transformFst Char.ToLower) |>  Seq.where (f textFrequency) |> Seq.sortBy fst |> Seq.map snd |> Vector.Build.SparseOfEnumerable
-        similarity textFrequencyVector languageFrequencyVector
+let checkCompatibility (similarity:Similarity) (languageFrequency: seq<char*float>) (textFrequency: seq<char*float>) =
```
Comment (gdziadkiewicz) [inline LanguageRecognition/Engine.fs]: @meoke This is the critical part. Please review it deeply.

Comment (meoke) [inline LanguageRecognition/Engine.fs]: @gdziadkiewicz Why did you use SparseVector in the toVector function? We won't get here many zero values (if any?), will we?

Comment (gdziadkiewicz) [inline LanguageRecognition/Engine.fs]: @meoke
You're right, it is guaranteed that there will be no zeros in that vector. I changed it.

**Comment by meoke** [comment]
+1

**Pull request merged**